sys/riscv/riscv/pmap.c

```diff
@@ -3411,6 +3449,15 @@ pmap_copy_page_tags(vm_page_t msrc, vm_page_t mdst)
 	vm_offset_t src = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(msrc));
 	vm_offset_t dst = PHYS_TO_DMAP(VM_PAGE_TO_PHYS(mdst));
 
+	/*
+	 * Copy across capability tracking flags since we might bypass the
+	 * TLB's capdirty emulation.  Count this as a dirtying event for
```

Comment:
1. MIPSism
2. This should be done wherever the equivalent for plain dirty is done (which needs to be handled otherwise there will be consistency issues)

Reply:
1.  Eh?  We still do TLB-driven capdirty emulation on RISC-V since not all TLBs write back, right?

2. Hm.  I could push this out to the callers of `pmap_copy_page_tags`, but I notice that `vm_fault_cow` merely marks `dst`, there pronounced `fs->first_m` *valid* rather than *dirty*.  That feels wrong, in the same way that `MADV_FREE` was wrong.  @bsdjhb?

Reply:
1. PTE not TLB.
2. I think we should (though if it makes you happier we can have a KASSERT here that if the source page is cap dirty then so is the destination page, provided the dirty marking happens _first_, otherwise you just have to hope)